lms/djangoapps/instructor_task/tasks_helper/utils.py

```diff
@@ -45,3 +54,41 @@ def tracker_emit(report_name):
     Emits a 'report.requested' event for the given report.
     """
     tracker.emit(REPORT_REQUESTED_EVENT_NAME, {"report_type": report_name, })
+
+
+class SensitiveMessageOnReports(object):
+    """
+    Check how to add the disclaimer new row in the CSV,
+    due to reports are generated of different ways, each one handles their own way:
+        1. Using the CSV library directly.
+        2. Using upload_csv_to_report_store function which in turn
+           uses store_rows of DjangoStorageReportStore class where is built the CSV.
+    """
+
+    def __init__(self):
+        self.display_msg = configuration_helpers.get_value('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', settings.FEATURES.get('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', False))
+
+    def csv_direct(self, writer):
+        """
+        Writes the row immediately in the CSV.
+        """
+        if self.display_msg:
+            msg = self.process_message()
+            encode = unicode(msg).encode('utf-8')
+            return writer.writerow([encode])
+
+    def with_report_store(self):
+        """
+        Return the string parsed in process_message. This string is passed
+        to upload_csv_to_report_store which decides if execute it or not.
+        """
+        return [self.process_message()]
+
+    def process_message(self):
+        """
+        Return the message parsed from template.
+        """
+        template_message = 'instructor/instructor_dashboard_2/sensitive_data_on_reports.txt'
+        message = render_to_string(template_message, None)
+        message = message.replace('"', '')
```

Comment:
Why do you need to remove the quotes? What happens if you don't remove them?

Reply:
I just tested and nothing happens, I just write this line thinking the quotes were appears in the report and I did not test without it.

Reply:
we can remove it then ;)